packages/ipfs-daemon/src/create-repo.ts

```diff
@@ -16,6 +19,95 @@ const notALock = {
   locked: () => false,
 };
 
+export enum StorageBackend {
+  DEFAULT,
+  S3,
+}
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace StorageBackend {
+  // eslint-disable-next-line no-inner-declarations
+  export function fromEnv(input: string | undefined) {
+    switch (String(input).toLowerCase()) {
+      case 'default':
+        return StorageBackend.DEFAULT;
+      case 's3':
+        return StorageBackend.S3;
+      default:
+        return StorageBackend.DEFAULT;
+    }
+  }
+}
+
+type Options = {
+  path: string;
+  localPrefix: string | undefined;
+  createIfMissing: boolean;
+  backends: Record<string, StorageBackend>;
+};
+
+type S3Options = {
+  bucket?: string;
+  region?: string;
+  accessKeyId?: string;
+  secretAccessKey?: string;
+};
+
+class NoBackendConfigError extends Error {
+  constructor(backend: never) {
+    super(`Expect config for ${backend} config`);
+  }
+}
+
+function localPath(prefix: string | undefined, location: string) {
+  if (prefix) {
+    return path.resolve(prefix, location.replace(/^\//, ''));
+  } else {
+    return location;
+  }
+}
+
+function LocalDatastoreFS(pathPrefix: string | undefined) {
+  return class LocalDatastoreFS extends DatastoreFS {
+    constructor(repoPath: string, options: any) {
+      super(localPath(pathPrefix, repoPath), options);
+    }
+  };
+}
+
+function LocalDatastoreLevel(pathPrefix: string | undefined) {
+  return class LocalDatastoreLevel extends DatastoreLevel {
+    constructor(repoPath: string, options: any) {
+      super(localPath(pathPrefix, repoPath), options);
+    }
+  };
+}
+
+function makeBackend(config: any, name: string, options: Options, s3: () => S3, defaultBackend: any) {
```

Comment:
Can we name the function `setIpfsRepoBackend`?
And rename the options param to `ipfsRepoOptions`?

It took me a bit to understand all these moving pieces and more descriptive names will make it quicker.

Reply:
Agree. I just omit `ipfs` then, as it is _all_ about IPFS.